### PR TITLE
Update requirements for 8.5.0

### PR DESCRIPTION
**Pull request by jakirkham**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -50,8 +50,7 @@ requirements:
     - python
     - setuptools
     - fastrlock >=0.3
-    - numpy >=1.16
-    - six >=1.9.0
+    - numpy >=1.15
```
Comment (jakirkham) [inline recipe/meta.yaml:53]: Looks like an older version of NumPy works here. Though maybe we had a reason for requiring a newer one 

https://github.com/cupy/cupy/blob/d49845a49f25f2d5675adf27b243aaf6bc2eb62f/setup.py#L44

File: recipe/meta.yaml [hunk 2]
```diff
@@ -50,8 +50,7 @@ requirements:
     - python
     - setuptools
     - fastrlock >=0.3
-    - numpy >=1.16
-    - six >=1.9.0
```
Comment (jakirkham) [inline recipe/meta.yaml:54]: This got dropped a while ago ( https://github.com/cupy/cupy/commit/8d923719ced31fdff4a42702d395ec87a7e98045 ), but I think we missed it. Anyways we can drop it now

File: recipe/meta.yaml [hunk 3]
```diff
@@ -40,7 +40,7 @@ requirements:
     - python
     - pip
     - setuptools
-    - cython >=0.24.0
+    - cython >=0.29.22
```
Comment (leofang) [inline recipe/meta.yaml:43]: Same nitpick as I made in another PR:
```suggestion
    - cython >=0.29.22,<3
```

File: recipe/meta.yaml (hunk 2 above)
Comment (leofang) [reply recipe/meta.yaml:54]: Good catch! Thanks.

File: recipe/meta.yaml (hunk 1 above)
Comment (leofang) [reply recipe/meta.yaml:53]: I think that line is more likely an oversight (cc: @kmaehashi). CuPy v8 dropped the support for v1.15 (https://github.com/cupy/cupy/pull/3802), see also https://docs.cupy.dev/en/v8.5.0/install.html#python-dependencies.
```suggestion
    - numpy >=1.16
```

**Review by leofang — COMMENTED** [review]
Thanks for the PRs, John! Like in the other PR, could you add this to `run_constrained`?
```
  - scipy >= 1.3
  - optuna >=2
```
Note the lowest supported SciPy version is different. ref: https://docs.cupy.dev/en/v8.5.0/install.html#python-dependencies.

**Review by leofang — APPROVED** [review]

**Comment by leofang** [comment]
LGTM, thanks John! Marking automerge...

**Pull request merged**